statev2/src/replication/log_store.rs

```diff
@@ -112,8 +112,44 @@ impl LogStore {
     // | Setters |
     // -----------
 
+    /// Apply a config change to the log store
+    pub fn apply_config_state(&self, change: ConfState) -> Result<(), ReplicationError> {
+        let tx = self.db.new_write_tx().map_err(ReplicationError::Storage)?;
+
+        let value = ProtoStorageWrapper(change);
+        tx.write(RAFT_METADATA_TABLE, &CONF_STATE_KEY.to_string(), &value)
+            .map_err(ReplicationError::Storage)?;
+
+        tx.commit().map_err(ReplicationError::Storage)
+    }
+
+    /// Apply a hard state to the log store
+    pub fn apply_hard_state(&self, state: HardState) -> Result<(), ReplicationError> {
+        let tx = self.db.new_write_tx().map_err(ReplicationError::Storage)?;
+
+        let value = ProtoStorageWrapper(state);
+        tx.write(RAFT_METADATA_TABLE, &HARD_STATE_KEY.to_string(), &value)
+            .map_err(ReplicationError::Storage)?;
+
+        tx.commit().map_err(ReplicationError::Storage)
+    }
+
+    /// Append entries to the raft log
+    pub fn append_log_entries(&self, entries: Vec<RaftEntry>) -> Result<(), ReplicationError> {
+        let tx = self.db.new_write_tx().map_err(ReplicationError::Storage)?;
+        for entry in entries.into_iter() {
+            let key = lsn_to_key(entry.index);
+            let value = ProtoStorageWrapper(entry);
+
+            tx.write(RAFT_LOGS_TABLE, &key, &value)
```

Comment:
thoughts on exposing a batch-write method on the `DbTxn`?

Reply:
Prefer the lean interface, we don't need a batch method